src/main/java/org/jabref/gui/openoffice/OOBibBaseConnect.java

```diff
@@ -0,0 +1,271 @@
+package org.jabref.gui.openoffice;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.jabref.gui.DialogService;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.openoffice.CreationException;
+import org.jabref.logic.openoffice.DocumentConnection;
+import org.jabref.logic.openoffice.NoDocumentException;
+
+import com.sun.star.comp.helper.BootstrapException;
+import com.sun.star.container.NoSuchElementException;
+import com.sun.star.container.XEnumeration;
+import com.sun.star.container.XEnumerationAccess;
+import com.sun.star.frame.XComponentLoader;
+import com.sun.star.frame.XDesktop;
+import com.sun.star.lang.WrappedTargetException;
+import com.sun.star.lang.XComponent;
+import com.sun.star.lang.XMultiComponentFactory;
+import com.sun.star.text.XTextDocument;
+import com.sun.star.uno.UnoRuntime;
+import com.sun.star.uno.XComponentContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class for manipulating the Bibliography of the currently started
+ * document in OpenOffice.
+ */
+class OOBibBaseConnect {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OOBibBaseConnect.class);
+
+    /* variables  */
+    private final DialogService dialogService;
+    private final XDesktop xDesktop;
+
+    /**
+     * Created when connected to a document.
+     *
+     * Cleared (to null) when we discover we lost the connection.
+     */
+    private DocumentConnection xDocumentConnection;
+
+    /*
+     * Constructor
+     */
+    public OOBibBaseConnect(Path loPath,
+                            DialogService dialogService)
+        throws
+        BootstrapException,
+        CreationException {
+
+        this.dialogService = dialogService;
+        this.xDesktop = simpleBootstrap(loPath);
+    }
+
+    /* *****************************
+     *
+     *  Establish connection
+     *
+     * *****************************/
```

Comment:
This is referring to the bootstrap process?

Reply:
Yes. No. The comment is likely from the time when it lived in OOBibBase and I tried to
organize the parts. I probably meant connection to a document (which needs the bootstrap,
but ...)

moved to the class description as

```
/**
 * Establish connection to a document opened in OpenOffice or LibreOffice.
 */
```